python/orca/src/bigdl/orca/data/shard.py

```diff
@@ -351,34 +351,23 @@ def merge_rows(iter):
                 else:
                     dtypes.append(v.dtype)
             if dict_of_batched_ndarray:
-                if num_partitions > self.rdd.getNumPartitions():
```

Comment:
Remove coalesce for SparkXShards of dict as well since it may also be the input of users.